### PR TITLE
Feat: Added Gradio support

### LINKED ISSUE #228: [Feat] Add gradio
Gradio can be used to share a simple interface for axolotl, particularly share=Public to let others test.

### PR DESCRIPTION
**Pull request by Stillerman**
You can start public Gradio UI by passing `--gradio` to the inference CLI.

```bash
accelerate launch -m axolotl.cli.inference examples/openllama-3b/config.yml --gradio
```

Closes #228

### CODE REVIEW

**Review by winglian — APPROVED** [review]
lgtm, I'll see if I can get this working locally and merge once I can validate. Thank you!

**Comment by winglian** [comment]
@Stillerman would you be able to run `pre-commit run --all-files` to fix the listing issues pls?

**Comment by Stillerman** [comment]
Yes I can do that by tomorrow morning

File: src/axolotl/cli/__init__.py [hunk 1]
```diff
@@ -153,6 +155,86 @@ def do_inference(
...
+            thread.start()
+
+            all_text = ""
+
+            for new_text in streamer:
+                all_text += new_text
+                yield all_text
+
+    demo = gr.Interface(fn=generate, inputs="text", outputs="text")
+    demo.launch(show_api=False, share=True)
```
Comment (winglian) [inline src/axolotl/cli/__init__.py]:
```suggestion
    demo.queue().launch(show_api=False, share=True)
```

Ran into this error: `ValueError: Queue needs to be enabled! You may get this error by either 1) passing a function that uses the yield keyword into an interface without enabling the queue or 2) defining an event that cancels another event without enabling the queue. Both can be solved by calling .queue() before 
.launch()`, the above change seems to fix it.

Comment (Stillerman) [reply src/axolotl/cli/__init__.py]: Weird that it didn't show up for me... Added it anyways and also a `--gradio_title` param

**Pull request merged**